content/courses/program-security/signer-auth.md

```diff
@@ -316,8 +321,20 @@ pub mod signer_authorization {
     }
 
     pub fn insecure_withdraw(ctx: Context<InsecureWithdraw>) -> Result<()> {
-        let amount = ctx.accounts.token_account.amount;
+          // Check if the amount is greater than zero
+    if amount == 0 {
+        return Err(ProgramError::InvalidArgument.into());
+    }
 
+    // Check if there are enough tokens in the account
+    if ctx.accounts.token_account.amount < amount {
+        return Err(ProgramError::InsufficientFunds.into());
+    }
+
+    // Check if the withdraw destination is different from the vault's token account
+    if ctx.accounts.token_account.key() == ctx.accounts.withdraw_destination.key() {
+        return Err(ProgramError::InvalidArgument.into());
+    }
         let seeds = &[b"vault".as_ref(), &[*ctx.bumps.get("vault").unwrap()]];
```

Comment:
Can ensure the code is properly formatted (4 spaces per CONTRIBUTING.md)? Looks like you're in one level too far left.